app/models/genre.rb

```diff
@@ -0,0 +1,37 @@
+class Genre < ApplicationRecord
+
+  include Models::Activable
+  include Sluggable
+
+  belongs_to :super_genre, class_name: 'Genre', optional: true
+  has_many :sub_genres, class_name: 'Genre', foreign_key: 'super_genre_id'
+
+  validates :title, uniqueness: { scope: :super_genre_id }, presence: true
+  validates :slug, presence: true, uniqueness: true
+  validate :single_level_nesting
+
+  def to_param
+    slug
+  end
+
+  private
+
+  def self.ransackable_attributes(auth_object = nil)
```

Comment:
Do we need this as private class method??

Reply:
writing class methods under private scope donot make them private, we can still call them explicity on the class